modules/javafx.graphics/src/main/native-glass/gtk/glass_key.cpp

```diff
@@ -345,4 +346,63 @@ JNIEXPORT jint JNICALL Java_com_sun_glass_ui_gtk_GtkApplication__1getKeyCodeForC
     return gdk_keyval_to_glass(keyval);
 }
 
+/*
+ * Function to determine whether the Xkb extention is available. This is a
+ * precaution against X protocol errors, although it should be available on all
+ * Linux systems.
+ */
+
+static Bool xkbInitialized = False;
+static Bool xkbAvailable = False;
+
+static Bool isXkbAvailable(Display *display) {
+    if (!xkbInitialized) {
+        int xkbMajor = XkbMajorVersion;
+        int xkbMinor = XkbMinorVersion;
+        xkbAvailable = XkbQueryExtension(display, NULL, NULL, NULL, &xkbMajor, &xkbMinor);
+        xkbInitialized = True;
+    }
+    return xkbAvailable;
+}
+
+/*
+ * Class:     com_sun_glass_ui_gtk_GtkApplication
+ * Method:    _isKeyLocked
+ * Signature: (I)I
+ */
+JNIEXPORT jint JNICALL Java_com_sun_glass_ui_gtk_GtkApplication__1isKeyLocked
+  (JNIEnv * env, jobject obj, jint keyCode)
+{
+    Display* display = gdk_x11_display_get_xdisplay(gdk_display_get_default());
+    if (!isXkbAvailable(display)) {
+        return com_sun_glass_events_KeyEvent_KEY_LOCK_UNKNOWN;
+    }
+
+    Atom atom = None;
```

Comment:
The documentation of [XInternAtom()](https://www.x.org/releases/X11R7.5/doc/man/man3/XInternAtom.3.html) does mention that this function returns an atom identifier associated with the provided atom name. So the variable name 'atom' does sound good enough. But would it be good to rename this variable to something like 'keyCodeAtom'. I am Ok either way.

Reply:
fixed